### PR TITLE
UXODS-253 tabs links style like btns & UXODS-279 disabled tab cursor

### PR DESCRIPTION
**Pull request by wc0599**
_Note: this Pull Request combines two tickets--UXODS-253 and UXODS-279_

# UXODS-253: Tabs links style like btns
All of the Tab stories currently only use buttons:

```html
<div class="jazz-tablist" role="tablist">
      <button role="tab" id="one">Tab Label A</button>
      <button role="tab" id="two" aria-selected="true">Tab Label B</button>
      <button role="tab" disabled>Disabled Button</button>
      <button role="tab">Groups</button>
      <button role="tab">Funding</button>
      <button role="tab">Expenses</button>
      <button role="tab" aria-controls="icons">Icon</button>
</div>
```
    
Need for two different navigation options (`<a>` vs `<button>`) for tabs in cases where each tab "navigates" to a different page:

```html
<div class="jazz-tablist" role="tablist">
      <a href="javascript:void(0)" role="tab" id="one">Tab Label A</a>
      <a href="javascript:void(0)" role="tab" id="two" aria-selected="true">Tab Label B</a>
      <span role="tab">Disabled Button</span>
      <a href="javascript:void(0)" role="tab">Groups</a>
      <a href="javascript:void(0)" role="tab">Funding</a>
      <a href="javascript:void(0)" role="tab">Expenses</a>
      <a href="javascript:void(0)" role="tab" aria-controls="icons">Icon</a>
</div>
```
- The component now assumes each __Tab__ component contains either all--`<button>`s or `<a>`tags (with the exception of _disabled_ tabs, in which case `<span>`s are used).

## Reason for Change
- Although [MDN recommends](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Tab_Role#best_practices) using `<button>`s for tab functionality, our use of the Angular framework would be better served to also use `<a>` tags in cases where Angular's `<Router />` components are used for navigating to different pages.

## Changed
- Storybook has been updated to show `Tab` component implemented in `<button>`s and `<a>`tags.

## Next Steps
- Update Unit Test files to reflect the newly added `<a>` tags.
- Update Design Systems documentation for Tabs component to describe uses for `<a>`tags (currently, the Design Systems docs only describe Tabs component using `<button>`.

# UXODS-279: Disabled tab cursor
Cursor value is currently set to `pointer` for any _disabled_ tabs.

## Changed
Cursor value is now set to `default` for any _disabled_ tabs.

### CODE REVIEW

File: src/scss/_tabs.scss [hunk 1]
```diff
@@ -5,10 +5,13 @@ $jazz-tab-border-size: 4;
...
   // bottom border serves as a rail across the entire set of tabs
-  border-bottom: px-to-rem($jazz-tab-border-size) solid jazz-theme(tablist-border-color);
+  border-bottom: px-to-rem($jazz-tab-border-size) solid
+    jazz-theme(tablist-border-color);
 }
 
-.jazz-tablist[role="tablist"] button[role='tab'] {
+.jazz-tablist[role="tablist"] button[role="tab"],
+.jazz-tablist[role="tablist"] a[role="tab"],
+.jazz-tablist[role="tablist"] span {
```
Comment (mzarillopu) [inline src/scss/_tabs.scss:14]: Should the span selector be scoped to [role="tab"] also?  If so, I wonder if the selector can be made more generic (*[role="tab"])?

**Review by mzarillopu — CHANGES_REQUESTED** [review]

File: src/js/components/tabs/tabs.links.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,619 @@
+import {TabsBehavior} from "./tabs";
+import {simulateKeyupEvent} from "../../../../jest-setup";
+
+describe('tab initialization behavior', () => {
+  it('should not throw an error when a selected tab button does not define what it controls', () => {
+    document.body.innerHTML = `
+        <div class="jazz-tablist" role="tablist">
+          <button role="tab">Tab Label 1</button>
```
Comment (rs16princeton) [inline src/js/components/tabs/tabs.links.spec.tsx:8]: The tests for this file should've been testing for the a tags (anchor tags) right? Did you forget to edit this file by accident?

Comment (wc0599) [reply src/js/components/tabs/tabs.links.spec.tsx:8]: Oh I thought adjusting the unit tests for the tabs was another ticket, and I was working on the tests now.

Should I add in the changes for the unit tests before merging this ticket?

File: src/scss/_tabs.scss (hunk 1 above)
Comment (wc0599) [reply src/scss/_tabs.scss:14]: Yes, the span selector can and probably should have the scoping `[role="tab"]` added.

And I also think a more generic selector `*[role="tab"]` would work, assuming that no other components, tags or elements in the page will also have the `role="tab"` property.

Comment (mzarillopu) [reply src/scss/_tabs.scss:14]: Keep in mind that the scope here is already within the ".jazz-tablist" class, so you don't have to worry about the rest of the page.  I am not sure if the "*" syntax that I provided is correct, so you will have to try it to make sure that it works as we want it to here.  Thanks!

**Review by rs16princeton — APPROVED** [review]

**Pull request merged**